### PR TITLE
Couple of fixes and tweaks to the R2R creation logic

### PR DESCRIPTION
**Pull request by fadimounir**
Changing crossgen commands to use response files.
Changing /Platform_Assemblies_Paths to a set of /r arguments
Using resolved files list as assembly references to crossgen (temp hack until issue 3110 is fixed)

### CODE REVIEW

**Comment by fadimounir** [comment]
@nguerrera PTAL

**Comment by fadimounir** [comment]
@nguerrera PTAL

File: src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs [hunk 1]
```diff
@@ -102,5 +220,25 @@ protected override int ExecuteTool(string pathToTool, string responseFileCommand
 
             return base.ExecuteTool(pathToTool, responseFileCommands, commandLineCommands);
         }
+
+        #region TEMP - DELETE BEFORE MERGE (used for repros now)
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs]: Note to delete this as you intended.

File: src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs [hunk 2]
```diff
@@ -80,18 +84,132 @@ protected override bool ValidateParameters()
...
+                {
+                    // Not a valid assembly file
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private bool IsReferenceAssembly(MetadataReader mdReader)
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs]: I know you're saying this is temporary, but why do we need to duplicate what the prepare step checks here? Why can't we handle building the right list in the prepare step?

**Review by nguerrera — CHANGES_REQUESTED** [review]

**Comment by fadimounir** [comment]
> I know you're saying this is temporary, but why do we need to duplicate what the prepare step checks here? Why can't we handle building the right list in the prepare step?

I don't really have a preference on where to put this. I was thinking it would be easier to fix this later by deleting a well defined piece of code rather than some scattered bits. It's just a matter of style really

**Comment by nguerrera** [comment]
> I don't really have a preference on where to put this. I was thinking it would be easier to fix this later by deleting a well defined piece of code rather than some scattered bits. It's just a matter of style really

OK, that's fine since this is just temporary until https://github.com/dotnet/sdk/issues/3110

File: src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs (hunk 1 above)
Comment (fadimounir) [reply src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs]: FWIW, filed https://github.com/Microsoft/msbuild/issues/4315. I really think this should be fixed in msbuild

**Review by nguerrera — APPROVED** [review]

**Pull request merged**